src/info/worker.ts

```diff
@@ -694,14 +644,89 @@ function activeGuidance(segments: SubSegment[][], tSegmentDuration: number,
-        moveToPos(coord);
+        // move to the point
+                // to avoid setting a large force at once
+                // when moving from the end of one segment to the beginning of another
+                // we linear interpolate the # of points
+                // to avoid setting large forces
+                // except in the case of setting the force from home pos
+                // which needs a larger force to compensate
+                if (currentSubSegmentPointIndex == 0 && !atHomePos()) {
+
+                  // case where we need to transition from segment to segment
+                  switch (transition) {
+                      case Transition.GetPoints: {
+                          // interpolate point data
+                          const vEndEffector = new Vector(convPosEE.x, convPosEE.y);
+                          upSampled = upsample([vEndEffector, coord], 6500);
+                          tHoldTimeSegToSeg = Date.now();
+                          transition = Transition.Move;
+                          break;
+                      }
+                      case Transition.Move: {
+                          // if we are done then end
+                          if (idx >= upSampled.length - 1) {
+                              transition = Transition.Rest;
+                              idx = 0;
+                              break;
+                          }
+                          else {
+                              // force reduction as we approach the threshold
+                              let k = 3;
+                              let coeff = 1;
+                              if (diff.mag() < 0.025) {
+                                  let unit = diff.unit();
+                                  const multiplier = 1 / (diff.mag() + 0.9);
+                                  const coeff = Math.min(1, 1 - ((unit.x * unit.y) * multiplier * multiplier));
+                              }
+                              // move to new point with the WaitTime refresh rate
+                              if (Date.now() - tHoldTimeSegToSeg > tWaitTime) {
+                                  moveToPos(upSampled[idx], k * coeff);
+                                  idx++;
+                                  tHoldTimeSegToSeg = Date.now();
+                              }
+                          }
+                          break;
+                      }
+                      case Transition.Rest: {
+                          force.set(0, 0);
+                          fEE.set(graphics_to_device(force));
+                          break;
+                      }
+                  }
+              }
+              // case where we need to transition from point to point in a segment
+              else {
+                moveToPos(coord);
+            }
       }
     }
   }
 }
 
+enum Transition {
+  GetPoints,
+  Move,
+  Rest
+}
+
+// for transition between segments
+let transition: Transition = Transition.GetPoints;
```

Comment:
i guess you need to move these variable declarations at the top somehere, I can see that they are being used in the lines above, where these will be undefined and the code may not work in an expected manner. Please check